frontend/src/metabase/query_builder/components/QuestionEmbedWidget/QuestionEmbedWidget.tsx

```diff
@@ -0,0 +1,66 @@
+import type { Card } from "metabase-types/api";
+import type { EmbedOptions } from "metabase-types/store";
+import type { ExportFormatType } from "metabase/dashboard/components/PublicLinkPopover/PublicLinkCopyPanel";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import * as Urls from "metabase/lib/urls";
+import { EmbedModal } from "metabase/public/components/widgets/EmbedModal";
+import EmbedModalContent from "metabase/public/components/widgets/EmbedModalContent";
+import { getMetadata } from "metabase/selectors/metadata";
+import { getCardUiParameters } from "metabase-lib/parameters/utils/cards";
+
+import {
+  createPublicLink,
+  deletePublicLink,
+  updateEnableEmbedding,
+  updateEmbeddingParams,
+} from "../../actions";
+
+type QuestionEmbedWidgetProps = {
+  className?: string;
+  card: Card;
+  onClose: () => void;
+};
+export const QuestionEmbedWidget = (props: QuestionEmbedWidgetProps) => {
+  const { className, card, onClose } = props;
+
+  const metadata = useSelector(getMetadata);
+
+  const dispatch = useDispatch();
+  const createPublicQuestionLink = () => dispatch(createPublicLink(card));
+  const disablePublicQuestionLink = () => dispatch(deletePublicLink(card));
+  const updateQuestionEnableEmbedding = (enableEmbedding: boolean) =>
+    dispatch(updateEnableEmbedding(card, enableEmbedding));
+  const updateQuestionEmbeddingParams = (embeddingParams: EmbedOptions) =>
+    dispatch(updateEmbeddingParams(card, embeddingParams));
+
+  const getPublicUrl = (
+    {
+      public_uuid,
+    }: {
+      public_uuid: string;
+    },
+    extension: ExportFormatType,
```

Comment:
I don't see you pass `extension` now and I don't think the extension is being used in the new design as well. As far as I know, `extension` is only for the public link, so this one doesn't have anything to do with public embed. Should it be removed?

You can also verify this from the old UI that we don't show extensions in public embed.

![image](https://github.com/metabase/metabase/assets/1937582/b891d93e-eac6-471d-a60b-2f5226b6db2e)

Reply:
I'm a bit confused. Are you going to remove the `extension` here? I still see it here.

Reply:
My bad, I thought you had just meant removing the prop and keeping the public url method, but that can be removed as well